pkg/apis/serving/v1/revision_types.go

```diff
@@ -129,13 +129,22 @@ type RevisionStatus struct {
 	// +optional
 	LogURL string `json:"logUrl,omitempty"`
 
-	// ImageDigest holds the resolved digest for the image specified
+	// DeprecatedImageDigest holds the resolved digest for the image specified
 	// within .Spec.Container.Image. The digest is resolved during the creation
 	// of Revision. This field holds the digest value regardless of whether
 	// a tag or digest was originally specified in the Container object. It
 	// may be empty if the image comes from a registry listed to skip resolution.
+	// DEPRECATED Use ImageDigests instead.
+	// TODO(savitaashture) Remove deprecatedImageDigest in 2 releases(0.16).
 	// +optional
-	ImageDigest string `json:"imageDigest,omitempty"`
+	DeprecatedImageDigest string `json:"imageDigest,omitempty"`
+
+	// ImageDigests holds the resolved digest for the specified images
+	// within .Spec.Container.Image. The digest is resolved during the creation
+	// of Revision. ImageDigests holds the digest for all the .Spec.Container.Image both serving and non serving.
+	// ref: http://bit.ly/image-digests
+	// +optional
+	ImageDigests map[string]string `json:"imageDigests,omitempty"`
```

Comment:
K8s style prefers `name/value` structs over maps.

Reply:
Do you mean we should use struct instead of map[string]string ?

Reply:
I think the idea is to have a list of Name/Value pair structs, similar to `Env: []EnvVar`

Reply:
> K8s style prefers name/value structs over maps.

Curious why?

Reply:
In general, it is related to patching strategies, but I think my preference is more about consistency with K8s APIs (since I'm not sure those strategies apply to CRDs).

Reply:
Also deserialize-serialize will yield different wire object — due to the order of iteration, which is not good in general.

Reply:
So shall i go ahead and change it to `Name/Value pair structs` ?

Reply:
I think so. Thanks.

Reply:
Updated ImageDigests to use struct instead of map

@mattmoor @markusthoemmes @vagababov @dprotaso Please take a look
Thank you

Reply:
So for posterity: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#lists-of-named-subobjects-preferred-over-maps
the way I read it: "pure maps", which I presume primitive to primitive are allowed and for example, see new endpointslices.endpoints.topology field.